### PR TITLE
Improve reward groups usage and presentation

### PR DESCRIPTION
**Pull request by AricStewart**

### CODE REVIEW

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 1]
```diff
@@ -254,34 +325,54 @@ export default {
...
-        } else if (column.type === String) {
-          result = a[column.value].localeCompare(b[column.value]) * sortDirection;
+      sortPrizes(this.records, column, sortDirection);
+      this.groups.forEach((group) => {
+        sortPrizes(group.items, column, sortDirection);
+      });
+    },
+    printGroupValue(record, column) {
+      if (column.value === 'value') {
+        if (this.groups[record.reward_group.id].valueLow == this.groups[record.reward_group.id].valueHigh) {
```
Comment (rkaiser-dev) [inline modules/volunteers/admin/sitesupport/giftTable.js:339]:
```suggestion
        if (this.groups[record.reward_group.id].valueLow === this.groups[record.reward_group.id].valueHigh) {
```

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 2]
```diff
@@ -254,34 +325,54 @@ export default {
...
-        if (a.reward_group !== null && b.reward_group === null) {return -1;}
-        if (b.reward_group !== null && a.reward_group === null) {return 1;}
-        if (b.reward_group === null && a.reward_group === null) {return 0;}
-        if (parseInt(b.reward_group.id) === parseInt(a.reward_group.id)) {return 0;}
-        return parseInt(a.reward_group.id) - parseInt(b.reward_group.id);
-      });
-    }
+      if (column.value === 'inventory') {
+        const str = 'Click to ' + ((this.groups[record.reward_group.id].open) ? 'close' : 'open');
+        return str;
```
Comment (rkaiser-dev) [inline modules/volunteers/admin/sitesupport/giftTable.js:358]: We could potentially simplify this with template literals:

```suggestion
        return `Click to ${this.groups[record.reward_group.id].open ? 'close' : 'open'}`;
```

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 3]
```diff
@@ -59,20 +90,22 @@ export default {
...
-    this.groupIndex = -1;
-    this.groupColorId = null;
     this.lastGroup = null;
     this.hoursRemaining = this.totalHours - this.hoursSpent;
   },
   methods: {
     load() {
+      this.records = [];
+      this.claims = [];
+      this.groups = [];
```
[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:107]: should not be.

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 4]
```diff
@@ -99,17 +167,9 @@ export default {
...
-                      record.acquired = record.acquired ? record.acquired + 1 : 1;
-
-                      // Assuming value is actually null
-                      if (record.reward_group !== null) {
-                        this.groupCount[record.reward_group.id] =
-                          this.groupCount.includes(record.reward_group.id) ? this.groupCount[record.reward_group.id] + 1 : 1;
-                      }
-                    }
-                    return record;
+                  this.records.map((record) => { mapEntry(record, entry, this.groups); });
```
[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:170]: It seems to be working fine. I changed to your code with no difference. So I will make your changes.

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 5]
```diff
@@ -99,17 +167,9 @@ export default {
...
+                  this.groups.forEach((group) => {
+                    group.items.map((record) => { mapEntry(record, entry, this.groups); });
```
[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:172]: Yup, but no issue with changing.

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 6]
```diff
@@ -173,36 +246,34 @@ export default {
...
+      if (this.$parent.userId && !this.acquirable(row) && row.type !== 'group') {
+        base = 'UI-disabled';
       }
       if (row.reward_group != null) {
-        if (row.reward_group.id != this.groupColorId) {
-          this.groupIndex += 1;
-          this.groupColorId = row.reward_group.id;
-        }
-        return base + GROUPSTYLES[this.groupIndex];
+        base = base.concat(' ',this.groups[row.reward_group.id].color);
```
[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:276]: easy change. no issue.

File: modules/volunteers/admin/sitesupport/giftTable.js [hunk 7]
```diff
@@ -254,34 +325,54 @@ export default {
...
+      });
+    },
+    printGroupValue(record, column) {
+      if (column.value === 'value') {
+        if (this.groups[record.reward_group.id].valueLow == this.groups[record.reward_group.id].valueHigh) {
+          return this.$parent.printHours(this.groups[record.reward_group.id].valueLow);
         }
+        return '<' + this.$parent.printHours(this.groups[record.reward_group.id].valueLow) + ' -- ' +
+          this.$parent.printHours(this.groups[record.reward_group.id].valueHigh) + '>';
+      }
```
[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:344]: Comes in with a line to long issue.  Is there a way to break across lines?

[the remark this answers is not in the document]
Comment (AricStewart) [reply modules/volunteers/admin/sitesupport/giftTable.js:344]: Looks like I can use '\'  so 

```
          return `<${this.$parent.printHours(this.groups[record.reward_group.id].valueLow)} -- \
          ${this.$parent.printHours(this.groups[record.reward_group.id].valueHigh)}>`;
```

looks like it works.

File: modules/volunteers/admin/sitesupport/giftTable.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (rkaiser-dev) [reply modules/volunteers/admin/sitesupport/giftTable.js:170]: Interesting, I wonder if there are side effects at play here. I'll make a note to explore more about it later on.

**Review by rkaiser-dev — APPROVED** [review]

**Pull request merged**